### ISSUE TITLE
Provide Tenant Certificates After Sql Server Startup

### ISSUE THREAD

**Issue by JeffSwenson** [cockroachdb/cockroach#77336]
To cut down own startup latency, Serverless pre allocates Kubernetes pods for cockroachdb sql servers. When a pod is created, instead of running cockroachdb, it runs a "sqlstarter" server. The sqlstarter server accepts a set of client certificates, writes them to the pod's storage, then starts the cockroachdb sql server.

Ideally it would be possible to start an unbound sql server. The sql server would wait to receive its tenant ID and certificates. Pushing this logic into the sql server cuts down on cold start latency. The cold start process does not need to load the cockroach process into memory and it allows the sql server to pre-initialize components like the logger.

**Comment by ajwerner** [comment]
We also need to provide the certificates when providing the tenant ID.

**Comment by knz** [comment]
> provide the certificates when providing the tenant ID.

This can be done by placing them in the certs dir before giving the tenant ID to the server -- they are only loaded on demand.

**Comment by ajwerner** [comment]
>> provide the certificates when providing the tenant ID.
>
> This can be done by placing them in the certs dir before giving the tenant ID to the server -- they are only loaded on demand.

Can you say more? My understanding is that the certificates embed the tenant ID. The process which would like to "stamp" the tenant onto the server process does not have disk access to the disk of the server process.

**Comment by knz** [comment]
How does the deployment work currently? I presume the "proper" tenant cert is also placed in the right position dynamically.

**Comment by ajwerner** [comment]
Yes, currently there's a process "sqlstarter" which receives an HTTP request with the certificates in the body in a zip and writes them to the appropriate place before exec'ing `cockroach`. A problem with this (the motivation for this issue) is that that exec call can take a long time (hundreds of milliseconds). 

Secondly, we don't have a great way to know when the server is ready to serve sql connections, so we poll, and in polling are exposed to more latency or cost.

**Comment by ajwerner** [comment]
Sorry, what I mean is that the process which __sends__ the HTTP request does not have access to the disk. We would like to remove the `sqlstarter` process which today __recieves__ the HTTP request (the "stamp") and replace it with logic inside cockroach, so that we can eliminate the latency due to loading the cockroach binary from disk and some of its other preliminary startup.

**Comment by knz** [comment]
The way I was seeing is that _sqlstarter would remain_ but merely alongside the crdb proecss, which would start at the same time / early. sqlstarted would talk to the crdb process running alongside it when the time came to set up the tenant id.

**Comment by ajwerner** [comment]
Alrighty, that'd be fine.
